### ISSUE TITLE
Feature Request : Tab Suspension

### ISSUE THREAD

**Issue by aerosol-can** [brave/brave-browser#17453]
#### I believe this feature will:
 *  ⚡ Provide a significant performance boost for:
     * Devices that don't have enough memory.
     *  For users with a workflow that needs a significant amount of tabs open.

#### Browsers with similar feature built-in :
*  Microsoft Edge

-------------------------------------------------

🙃 Side-Note: I do know about the extensions that provide similar functionalities. However, most of these extensions either aren't open source or trustable. Insecure Data Storage, Data Collection and Tracking are also somethings I look out for. Being built-in and backed by the Brave Browser, this will be much more trustable. 

Would love to see it on the new release, Thank you for your good work 😊.

**Comment by bsclifton** [comment]
+1 from @clicky6 via https://github.com/brave/brave-browser/issues/17827
> ![image](https://user-images.githubusercontent.com/89839714/131794888-339cbb7b-5cb1-4939-9b6c-679a2608074c.png)
>
> Add a **sleeping tab** feature in Brave like it is present in Microsoft Edge. It helps to reduce consumption of system resources by putting tabs to sleep after a certain amount of time described by the user.

**Comment by 0xGitGuy** [comment]
Would love to see this. The Great Suspender turned out to be a disaster and it's tough for me to trust extensions with such permissions. Native browser support would be great!

**Comment by BunnyTheLifeguard** [comment]
So the current implementation only discards when system memory is running low, right? 🤔
https://developers.google.com/web/updates/2015/09/tab-discarding
https://www.chromium.org/chromium-os/chromiumos-design-docs/tab-discarding-and-reloading

There is also a fork of the great suspender extension before it introduced malware:
https://github.com/gioxx/MarvellousSuspender

Maybe the Brave team can take inspiration from this to integrate this natively?

**Comment by litesam** [comment]
Would really love this feature implemented tbh... I have like ~15 tabs opened and the pain when I revisit my **pinned mail tab** just to see it reload is too immense. Atleast implement this feature for like 3 tabs which are pinned 😭.

**Comment by DarthJahus** [comment]
This feature is useful on tablets / 2-in-1 devices. Would be great if you added it to Brave.

**Comment by Cabus1982** [comment]
This feature would place Brave among the best browsers. Even Google Chrome already adopted this feature, imagine that!

**Comment by dawgama** [comment]
I'm amazed too.

**Comment by Sa-Ja-Di** [comment]
@usgoose @dawgama   Well, Brave is a minimalist browser without much customization and options, as it's main feature are Crypto related.

If you want a browser which is more about options and features (side tabs, multi-functional side bar, design choice of round/square tabs/corners, Title bar and buttons add/removal - and you want all that implemented in a native way...

Then Microsoft Edge is your browser of choice... or Vivaldi... if you can swallow the fact that it is basically a "fake browser" - Chromium with an app on top

The choice is yours - Happily using Edge now, as Brave will never satisfy what i expect as user from a browser company.

**Comment by vrravalos** [comment]
Yes, I would like to have this option too in Brave. In the meantime I'm evaluating this 'similar' open source web extension called Snoozz: https://github.com/rohanb10/snoozz-tab-snoozing ;-)

**Comment by clicky6** [comment]
> @usgoose @dawgama Well, Brave is a minimalist browser without much customization and options, as it's main feature are Crypto related.
> 
> If you want a browser which is more about options and features (side tabs, multi-functional side bar, design choice of round/square tabs/corners, Title bar and buttons add/removal - and you want all that implemented in a native way...
> 
> Then Microsoft Edge is your browser of choice... or Vivaldi... if you can swallow the fact that it is basically a "fake browser" - Chromium with an app on top
> 
> The choice is yours - Happily using Edge now, as Brave will never satisfy what i expect as user from a browser company.

The only issue with Edge is its closed-source nature and data collection practices as it's Microsoft,what do you expect...

**Comment by prbhtkumr** [comment]
although yes, the memory saver feature has been implemented. but it would be nice to have a way to discard tabs manually as well (maybe an option in the tab context menu), won't you agree?

**Comment by vellak8081** [comment]
If there was a way to prevent tabs from being auto un-discarded on visit/focus, that would be great - and would fully replace the functionality of the best tab suspend/sleep extensions.

I currently have ~1400 tabs in my browser session (yes, I know....), and I'm managing that with The Marvelous suspender, but:

1) it doesn't know about tab groups, which is fine as long as you don't use it as a session manager
2) it's still manifest V2 and not one of the sanctioned manifest v2 extensions brave offers to download/keep

**Comment by crogonint** [comment]
I'd like to light a fire under this feature request. Last night I spent 2 hours before bedtime clicking "more" to get to the bottom of a rather lengthy page. It took so long, I was falling asleep clicking the button. Might have been three hours, I dunno. So I got up this morning to start reviewing the content in chronological order, opened up my laptop, switched back to that tab, and it instantly reloaded. 

I've spent the last hour or so digging in to the WHY this happened, because I am seriously pissed that I have to do that all over again. Not only that, but now I have to deal with it EVERY time I want to review or read a lengthy web page?? NOT acceptable! I've been researching the problem, and there are tons of people that are pissed off about THIS specific problem. A lot of them say they're going to switch browsers, rather than deal with the issue.

That tells me that THIS feature should have been repaired YEARS ago. WHY is it still sitting here ignored?? Get the ball rolling on this and make it block something relatively important. Ideally, set it up so that it can be backported in to Chromium, because the entire concept is anti-user based. Still, I love Brave, and if I have to never use Chrome again.. oh well! Please just give me an integrated solution, that I know is safe and privacy conscious. 

We don't need any weird freeze / suspend features added, either, like some have asked. Just dig in to the code and disable the auto-reload/discard feature. I mean, ideally, inject a toggle in there so we can turn the feature on, if we WANT to enable it for some insane reason. Ideally, just add the feature back where we right-click the tab to auto-reload every so many minutes/hours. It wasn't broken before, and Chromium should have just left it the hell alone, when it worked correctly.

Thank you SO SO much for efforts, I don't know WHAT we would do without the Brave Browser and DuckDuckGo to lean on to help us stay privacy focused. You all are the best of the best, in my book! :)

**Comment by crogonint** [comment]
This web admin is one of a handful of people that seems to have discovered evidence that there is a Brave bug that discards a page after 1-2 minutes. Was that ever investigated?
https://community.brave.com/t/stop-refreshing-not-working-page/152775

**Comment by crogonint** [comment]
Another, 10 minutes:
https://community.brave.com/t/killing-inactive-tabs/243281
apparently there's a duplicate request here:
https://github.com/brave/brave-browser/issues/15664

**Comment by Brave-Matt** [comment]
@rebron I think this is worth considering. We get a decent amount of requests for this:
https://community.brave.app/t/force-a-tab-to-be-inactive-discarded-from-right-click-context-menu/488140
https://community.brave.app/t/new-feature-request-manual-tab-inactive-active-toggle/598032/15
https://community.brave.app/t/please-add-manual-inactivate-tab/585928/4
https://community.brave.app/t/is-built-in-tab-suspension-sleep-snooze-freeze-etc-in-the-works/200160

Worth noting (if not already mentioned) that Firefox has this feature easily accessible from the context menu when right-clicking on a tab:

<img width="467" height="267" alt="Image" src="https://github.com/user-attachments/assets/838bb3a0-a396-455c-bd92-c750beab348d" />

**Comment by crogonint** [comment]
I'm not much of one for keyboard shortcuts. A white list to retain the cache for certain websites is a start, but can we get an icon to toggle on, to add websites from the toolbar, instead of digging in to the menu every time we want to add something?

**Comment by BeyremHF** [comment]
I actually made an extension for this. Check it out: https://github.com/BeyremHF/Sleep-Tab
